src/main/java/org/opensearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -1169,7 +1166,7 @@ private void detectAnomaly(
                 internalListener,
                 anomalyResults,
                 resultIndex,
-                () -> injectSecurity.close()
+                null
```

Comment:
Can we remove this parameter?

Reply:
That was my initial gut feel, as this is the only known usage of that method so it possibly *could* be removed ...

But the method signature is more general with a `CheckedRunnable<?> runBefore` to support chaining result requests. In this particular usage, we aren't chaining a request, so `null` terminates the (potential) sequence of Runnables.  I figured it does no harm to leave it as is.